autouri/autouri.py

```diff
@@ -319,8 +334,31 @@ def rm(self, no_lock=False):
         """
         with self.get_lock(no_lock=no_lock) as lock:
             self._rm()
+            logger.info('rm: {uri}'.format(uri=self._uri))
         return
 
+    def rmdir(self, dry_run=False, num_threads=DEFAULT_NUM_THREADS, no_lock=False):
+        """Recursively delete all files on a directory.
+        Use it at your own risk. This method is multi-threaded.
+
+        Args:
+            dry_run:
+                Dry-run.
+            num_threads:
+                Number of threads for deletion.
+        """
+        files = self.find_all_files()
+        if dry_run:
+            for uri in files:
+                logger.info('rm (dry-run): {uri}'.format(uri=uri))
+            return
+        num_files = len(files)
+        thread_ids = [i % num_threads for i in range(num_files)]
+
+        args = list(zip(files, thread_ids))
+        with multiprocessing.Pool(num_threads) as p:
+            p.map(autouri_rm, args)
```

Comment:
Might want to consider using `starmap`. Then the wrapper `autouri_rm` can take the two arguments separately.

Reply:
Fixed as you suggested